### ISSUE TITLE
Inventory Bug

### ISSUE THREAD

**Issue by elthran** [elthran/RPG-Game#209]
I died fighting a monster, went back to the barrracks and everything was weird. Then started a battle and inventory CMD crashed it for some reason.

![test](https://cloud.githubusercontent.com/assets/11773722/26713103/6c3216c8-479d-11e7-8e9e-ec15e6b6423a.jpg)

**Comment by klondikemarlen** [comment]
I will try and remember to write a test suite for:
`for item in hero.inventory` 
to make sure it always returns a list.
Though .... I thought that .iter() was supposed to return a list. Maybe It is supposed to be a generator.

It might be as simple as tweaking:
```
def __iter__(self):
        """Return a list of _all_ items in this inventory.
        
        Untested!
        Use: 
        for item in inventory:
            print(item)
        """

return [getattr(self, name) for name in Inventory.all_slot_names]
```

and making the return:
`return (getattr(self, name) for name in Inventory.all_slot_names)`
Note the parentheses instead of square brackets.

**Comment by klondikemarlen** [comment]
Fixed in branch issue212 commit c352089f3a5 pull request #213. I still haven't written a test suite to prove for sure that it works though.
